File: vulcanus/src/main/java/com/merkury/vulcanus/model/entities/Comment.java
package com.merkury.vulcanus.model.entities;

import com.merkury.vulcanus.model.interfaces.Votable;
import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.SuperBuilder;

import java.time.LocalDateTime;

@MappedSuperclass
@Data
@SuperBuilder
@NoArgsConstructor
@AllArgsConstructor
public abstract class Comment implements Votable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Builder.Default
    private Integer upVotes = 0;
    @Builder.Default
    private Integer downVotes = 0;

    @Builder.Default
    private LocalDateTime publishDate = LocalDateTime.now();

    @ManyToOne
    @EqualsAndHashCode.Exclude
    @ToString.Exclude
    @JoinColumn(name = "author_id")
    private UserEntity author;
}

File: venus/src/model/interface/comment/comment.ts
export default interface Comment {
  id: number;
  text: string;
  rating: number;
  upVotes: number;
  downVotes: number;
  publishDate: string;
  author: string;
  isAuthor: boolean;
  isUpVoted: boolean;
  isDownVoted: boolean;
}
